### PR TITLE
add go mod support. added makefile.

### PR DESCRIPTION
**Pull request by sarge**
## Proposed change

Add support for go modules #29 

Modules work well, easiest to not build them within your current GOPATH. Added a makefile to helpout with the build process.

** There is a problem with Travis CI, Coveralls and Go Modules - unsure of the resolution at this stage.

## Types of changes

What types of changes is this pull request introducing to flottbot? _Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

You can fill this out after creating your PR. _Put an `x` in the boxes that apply_

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by pa3ng** [comment]
Hey @sarge appreciate the PR! I was able to run your Make file with the generated `go.mod`/`go.sum` files and successfully built the binary 👍 

Just a quick question: How exactly did you generate the `go.mod`/`go.sum` files for your setup?

**Comment by sarge** [comment]
@pa3ng I used `go mod init flottbot` which inspects the code and possibly the dep files to generate the `go.mod/go.sum` files.
This actually introduced a bug. The mod file contained references to github.com/target/flottbot/... so would actually download the source from github on build rather than using the local source.

I should have done `go mod init github.com/target/flottbot` this tell go mod that these modules are provided by this package and uses the source from here rather pulling from github. This is why the code coverage wasn't working too. This is now fixed with the latest commit.

**Comment by sarge** [comment]
@wass3r Thanks for the suggestions I have cleaned up the ```.travis.yml``` and the contributing guide. The folder ```config-example``` is no longer be called ```config```. I think this means I don't get the same result as expected in the guide, i.e. the bot is running having parsed all rules. Might leave it to you to polish the new user on boarding process.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -13,22 +13,24 @@ To ensure that all developers follow the same guidelines for development, we hav
...
+git clone git@github.com:target/flottbot.git somepath/src/github.com/target/flottbot
 ```
 
-- Build the project dependencies using [dep](https://github.com/golang/dep):
+- Build the project:
 
 ```sh
 # Change into the project directory
-cd $GOPATH/src/github.com/target/flottbot
+cd somepath/src/github.com/target/flottbot
```
Comment (JordanSussman) [inline .github/CONTRIBUTING.md]: Shouldn't this just be `cd somepath/flottbot`?

**Comment by JordanSussman** [comment]
LGTM besides my one minor comment. 

Reminder to self - Looks like we will also need to update the [Dockerfiles](https://github.com/target/flottbot/tree/master/docker) once this is merged too. I created https://github.com/target/flottbot/issues/103 to track that progress.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (sarge) [reply .github/CONTRIBUTING.md]: Good suggestion, I have updated, thanks.

**Comment by wass3r** [comment]
thanks @sarge et al!

**Pull request merged**